src/gc/gc.cpp

```diff
@@ -26685,7 +26683,7 @@ void gc_heap::background_grow_c_mark_list()
         assert (new_c_mark_list);
         memcpy (new_c_mark_list, c_mark_list, c_mark_list_length*sizeof(uint8_t*));
         c_mark_list_length = c_mark_list_length*2;
-        delete c_mark_list;
+        delete [] c_mark_list;
```

Comment:
@Maoni0 I haven't looked at valgrind errors from running coreclr yet (see #10800).  I was speaking from experience.  I fixed several dozen of these valgrind errors on previous projects.  In practice I think it makes very little difference which delete is called, but the spec allows the two news to allocate on different heaps.  And the different deletes to be incompatible.